src/Controls/src/Core/Platform/iOS/Extensions/FormattedStringExtensions.cs

```diff
@@ -124,5 +127,129 @@ public static class FormattedStringExtensions
 
 			return attrString;
 		}
+
+		// TODO: Make public for .NET 8
+		internal static void RecalculateSpanPositions(this UILabel control, Label element)
+		{
+			if (element == null)
+				return;
+
+			if (element.TextType == TextType.Html)
+				return;
+
+			if (element?.FormattedText?.Spans == null
+				|| element.FormattedText.Spans.Count == 0)
+				return;
+
+			var finalSize = control.Frame;
+
+			if (finalSize.Width <= 0 || finalSize.Height <= 0)
+				return;
+
+			var inline = control.AttributedText;
+
+			if (inline == null)
+				return;
+
+			var range = new NSRange(0, inline.Length);
+
+			NSTextStorage textStorage = new NSTextStorage();
+			textStorage.SetString(inline);
+
+			var layoutManager = new NSLayoutManager();
+			textStorage.AddLayoutManager(layoutManager);
+
+			var textContainer = new NSTextContainer(size: finalSize.Size)
+			{
+				LineFragmentPadding = 0
+			};
+
+			layoutManager.AddTextContainer(textContainer);
+
+			var labelWidth = finalSize.Width;
+
+			var currentLocation = 0;
+
+			for (int i = 0; i < element.FormattedText.Spans.Count; i++)
```

Comment:
I might be missing something, but is it 'i' use elsewhere in the loop, can we move to a foreach?

Reply:
`Spans` is an `IList<Span>`, so `foreach` will allocate an enumerator. The `for` loop is a little less readable, but better for performance in this case.